src/averageAges.js

```diff
@@ -55,7 +63,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const onlyWomen = people.filter(women => (
+    women.sex === 'f'
+  ));
+
+  const onlyChildren = people.filter(children => (
+    onlyWomen.some(mother => children.mother === mother.name
+      && (onlyWithSon ? children.sex === 'm' : true))));
+
+  return onlyChildren.reduce((total, child) => (
+    total + (child.born - onlyWomen.find(women => (
+      women.name === child.mother
+    )).born)), 0) / onlyChildren.length;
```

Comment:
you can use people and delete get onlyWomen (-1 cycle). Also, your code is hard to read (

```suggestion
  const totalDifference = onlyChildren.reduce((total, child) => {
    const mother = people.find(
      women => women.name === child.mother,
    );
  
     return total + (child.born - women.born);
  }, 0);
    
  return totalDifference / onlyChildren.length;
```